gratipay/models/participant/email.py

```diff
@@ -106,6 +103,16 @@ def validate_email_verification_request(self, c, email, *packages):
         if not all(email in p.emails for p in packages):
             raise EmailNotOnFile()
 
+        unverified_email = c.one("""
+            SELECT address
```

Comment:
> Turns out it makes the most UX sense to disallow removing emails when verification is pending, to properly communicate the reality that the pending verification still exists.

This is also a security risk, as far as I can tell. If we don't allow canceling verification requests - a mistake in entering their email could result in a malicious user taking over their account.